### DISCUSSION TITLE
Sit-Test environment fails because of dependencies

### DISCUSSION THREAD

**Discussion by vramanufidato** [gruntwork-io/knowledge-base#639] [Help]
Hi Team,

Currently we are setting up sit-test and making changes for eks cluster but it is failing because of dependencies. We have placed the required modules as well that refers but still facing the errors.

Running command "terragrunt plan-all --terragrunt-non-interactive -input=false --terragrunt-modules-that-include=../sit-test/<reg>/sit/networking/vpc/terrgrunt.hcl"
1115[ecs-deploy-runner][2023-01-19T07:23:34+0000] time=2023-01-19T07:23:34Z level=warning msg='plan-all' is deprecated. Running 'terragrunt run-all plan -input=false' instead. Please update your workflows to use 'terragrunt run-all plan -input=false', as 'plan-all' may be removed in the future!
1116[ecs-deploy-runner][2023-01-19T07:23:36+0000] time=2023-01-19T07:23:36Z level=error msg=Error processing module at '/tmp/tmpor6_2d04/sit-ibmb/<reg>/sit/networking/vpc/terragrunt.hcl'. How this module was found: dependency of module at '/tmp/tmpor6_2d04/sit/<reg>/sit/services/eks-cluster'. Underlying error: Error reading file at path /tmp/tmpor6_2d04/sit-test/<reg>/sit/networking/vpc/terragrunt.hcl: open /tmp/tmpor6_2d04/sit-test/<reg>/sit/networking/vpc/terragrunt.hcl: no such file or directory
1117[ecs-deploy-runner][2023-01-19T07:23:36+0000] time=2023-01-19T07:23:36Z level=error msg=Unable to determine underlying exit code, so Terragrunt will exit with error code 1
1118[ecs-deploy-runner][2023-01-19T07:23:36+0000] [INFO] [infrastructure-deploy-script] 2023-01-19 07:23:36  "terragrunt plan-all" exited with code 1
1119[ecs-deploy-runner][2023-01-19T07:23:36+0000] Error: Could not run "terragrunt plan-all"
1120[ecs-deploy-runner][2023-01-19T07:23:36+0000] exit status 1
1121[ecs-deploy-runner][2023-01-19T07:23:36+0000] ERROR: exit status 1
1122ERROR: Deployment failed
1123xargs: bash: exited with status 255; aborting

---

<ins datetime="2023-01-19T07:37:58Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109810">Tracked in ticket #109810</a></p>
</ins>

**Comment by josh-padnick** [comment]
It sounds like your module that Terragrunt downloaded to `/tmp/tmpor6_2d04/sit-ibmb//sit/networking/vpc/terragrunt.hcl` is failing when it tried to read `/tmp/tmpor6_2d04/sit-test//sit/networking/vpc/terragrunt.hcl`. I do note that the diff between these two paths is `sit-ibmb` versus `sit-test`. Perhaps one of your Terragrunt files is referencing a file that exists in the `sit-test` environment but not the `sit-ibmb` environment?

Separately, using `run-all plan` is a nice convenience, but if you're hitting issues you should run `terragrunt plan` on an individual folder.

Since this error is occurring in the context of Gruntwork Pipelines, I recommend manually running these commands on your local computer to reproduce the error so that you can easily debug it there.
